src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/SelfcareApiServiceImpl.scala

```diff
@@ -0,0 +1,54 @@
+package it.pagopa.interop.backendforfrontend.api.impl
+
+import it.pagopa.interop.commons.utils.TypeConversions._
+import com.typesafe.scalalogging.{Logger, LoggerTakingImplicit}
+import it.pagopa.interop.backendforfrontend.api.SelfcareApiService
+import it.pagopa.interop.backendforfrontend.service.{SelfcareClientService, TenantProcessService}
+import it.pagopa.interop.backendforfrontend.model._
+import it.pagopa.interop.commons.logging.{CanLogContextFields, ContextFieldsToLog}
+import it.pagopa.interop.backendforfrontend.error.Handlers.handleError
+import it.pagopa.interop.commons.utils.AkkaUtils._
+import it.pagopa.interop.backendforfrontend.error.BFFErrors._
+import it.pagopa.interop.backendforfrontend.service.types.SelfcareClientTypes._
+
+import akka.http.scaladsl.marshalling.ToEntityMarshaller
+import akka.http.scaladsl.server.Directives.onComplete
+import akka.http.scaladsl.server.Route
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Success
+
+final case class SelfcareApiServiceImpl(
+  selfcareClientService: SelfcareClientService,
+  tenantProcessService: TenantProcessService
+)(implicit ec: ExecutionContext)
+    extends SelfcareApiService {
+
+  private implicit val logger: LoggerTakingImplicit[ContextFieldsToLog] =
+    Logger.takingImplicit[ContextFieldsToLog](this.getClass)
+
+  override def getInstitutionUserProducts()(implicit
+    contexts: Seq[(String, String)],
+    toEntityMarshallerSelfcareProductarray: ToEntityMarshaller[Seq[SelfcareProduct]],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem]
+  ): Route = {
+    val result: Future[Seq[SelfcareProduct]] =
+      for {
+        userUuid         <- getUidFutureUUID(contexts)
+        organizationUuId <- getOrganizationIdFutureUUID(contexts)
+        _ = logger.info(
+          s"Retrieving products for institution ${organizationUuId.toString} of user ${userUuid.toString}"
+        )
```

Comment:
probably not required as it is already logged [here](https://github.com/pagopa/interop-be-backend-for-frontend/pull/142/files#diff-8bea4c3bd97f3aed0a0edd0088ec68faf5d088e72424b78f4d9acf0c174ecdccR33)
```suggestion
```